### PR TITLE
Move mapper to a package

### PR DESCRIPTION
**Pull request by matthiasr**
This allows the mapping logic and configuration format to be re-used by
the graphite exporter.

Enables prometheus/graphite_exporter#37, cf. https://github.com/prometheus/graphite_exporter/pull/52#issuecomment-412324849

Signed-off-by: Matthias Rampke <mr@soundcloud.com>

### CODE REVIEW

File: mapper/telemetry.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+package mapper
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	mappingsCount = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "statsd_exporter_loaded_mappings",
```
Comment (matthiasr) [inline mapper/telemetry.go]: i am not sure what to do about this. I could change the metric name to be package- not exporter-specific, or change the constructor to get it injected from the main package.

Comment (travisgroth) [reply mapper/telemetry.go]: I'm thinking return the value during load, or keep track of it as a regular value internally.  The caller can then fetch the count and construct the metric however it wants.  This is probably cleaner than creating a new metric in what is effectively a config and parsing package.

In the statsd_exporter case, the prometheus metric would move into main/telemetry and you'd update it whenever the file is loaded.

**Comment by travisgroth** [comment]
Should probably tuck this all into a top level pkg/ directory since the intent is to allow the mapper to be consumed by semi-external projects.

File: mapper/telemetry.go (hunk 1 above)
Comment (matthiasr) [reply mapper/telemetry.go]: Hmm I don't like the scrape-time plumbing that this implies in every user of the package, and that would need to update with every change of telemetry. The package also already knows about Prometheus and imports the client library, so I think we can let main just pass in the Gauge. That's very flexible for users of the package, but also simple if you just want the damn metrics to work.

**Comment by matthiasr** [comment]
@travisgroth I moved everything under pkg/ and we're no longer initializing the gauge from within the library. PTAL!

File: mapper/telemetry.go (hunk 1 above)
Comment (travisgroth) [reply mapper/telemetry.go]: Makes sense to me 👍

**Pull request merged**